src/main/java/com/google/devtools/build/lib/bazel/repository/downloader/DownloadManager.java

```diff
@@ -194,6 +199,10 @@ public Path download(
       }
     }
 
+    if (disableDownload) {
```

Comment:
Do I understand correctly that this works because we don't provide any built-in repositories anymore that do downloading and we delegate everything to Starlark code instead?

Reply:
This only works for repo rule which uses `repository_ctx.download`.